x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/OperatorOnlyRegistry.java

```diff
@@ -7,98 +7,39 @@
      */
-    public OperatorPrivilegesViolation check(String action, TransportRequest request) {
-        if (SIMPLE_ACTIONS.contains(action)) {
-            return () -> "action [" + action + "]";
-        } else if (ClusterUpdateSettingsAction.NAME.equals(action)) {
-            assert request instanceof ClusterUpdateSettingsRequest;
-            return checkClusterUpdateSettings((ClusterUpdateSettingsRequest) request);
-        } else {
-            return null;
-        }
-    }
+    OperatorPrivilegesViolation check(String action, TransportRequest request);
 
-    private OperatorPrivilegesViolation checkClusterUpdateSettings(ClusterUpdateSettingsRequest request) {
-        List<String> operatorOnlySettingKeys = Stream.concat(
-            request.transientSettings().keySet().stream(),
-            request.persistentSettings().keySet().stream()
-        ).filter(k -> {
-            final Setting<?> setting = clusterSettings.get(k);
-            return setting != null && setting.isOperatorOnly();
-        }).toList();
-        if (false == operatorOnlySettingKeys.isEmpty()) {
-            return () -> (operatorOnlySettingKeys.size() == 1 ? "setting" : "settings")
-                + " ["
-                + Strings.collectionToDelimitedString(operatorOnlySettingKeys, ",")
-                + "]";
-        } else {
-            return null;
-        }
-    }
+    /**
+     * Checks to see if a given {@link RestHandler} is subject to restrictions. This method may return a {@link RestResponse} if the
+     * request is restricted due to operator privileges. Null will be returned if there are no restrictions. Callers must short-circuit
+     * the request and respond back to the client with the given {@link RestResponse} without ever calling the {@link RestHandler}.
+     * If null is returned the caller must proceed as normal and call the {@link RestHandler}.
+     * @param restHandler The {@link RestHandler} to check for any restrictions
+     * @return null if no restrictions should be enforced, {@link RestResponse} if the request is restricted
+     */
+    RestResponse checkRestFull(RestHandler restHandler);
+
+    /**
+     * Checks to see if a given {@link RestHandler} is subject to partial restrictions. A partial restriction still allows the request
+     * to proceed but will update/rewrite the provived {@link RestRequest} such that when the {@link RestHandler} is called the result
+     * is variant (likley a restricted view) of the response provided. The caller must use the returned {@link RestRequest} when calling
+     * the {@link RestHandler}
+     * @param restHandler The {@link RestHandler} to check for any restrictions
+     * @param restRequest The {@link RestRequest} to check for any restrictions
+     * @return {@link RestRequest} to use when calling the {@link RestHandler}, can be same or different object as the passed in parameter
+     */
+    RestRequest checkRestPartial(RestHandler restHandler, RestRequest restRequest);
```

Comment:
similarly I wonder if a change of signature would be good here?
something that would indicate that this method might return a different, 'enriched' RestRequest'
how about
`RestRequest maybeResponseFilteringRequest(..)` ?

naming is hard..

Reply:
naming is hard :)  .. however, I would like to keep the vocabulary of "fully restricted" and "partially restricted". For example we have other capabilities to filter responses that are not related to operator privileges.